plugins/out_azure_kusto/azure_kusto.c

```diff
@@ -0,0 +1,509 @@
+            continue;
+        }
+
+        /* Get timestamp and object */
+        flb_time_pop_from_msgpack(&tm, &result, &obj);
+
+        map_size = 1;
+        if (ctx->include_time_key == FLB_TRUE) {
+            map_size++;
+        }
+
+        if (ctx->include_tag_key == FLB_TRUE) {
+            map_size++;
+        }
+
+        msgpack_pack_map(&mp_pck, map_size);
+
+        /* include_time_key */
+        if (ctx->include_time_key == FLB_TRUE) {
+            msgpack_pack_str(&mp_pck, flb_sds_len(ctx->time_key));
+            msgpack_pack_str_body(&mp_pck, ctx->time_key, flb_sds_len(ctx->time_key));
+
+            /* Append the time value as ISO 8601 */
+            gmtime_r(&tm.tm.tv_sec, &tms);
+            s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                         FLB_PACK_JSON_DATE_ISO8601_FMT, &tms);
+
+            len = snprintf(time_formatted + s, sizeof(time_formatted) - 1 - s,
+                           ".%03" PRIu64 "Z", (uint64_t)tm.tm.tv_nsec / 1000000);
+            s += len;
+            msgpack_pack_str(&mp_pck, s);
+            msgpack_pack_str_body(&mp_pck, time_formatted, s);
+        }
+
+        /* include_tag_key */
+        if (ctx->include_tag_key == FLB_TRUE) {
+            msgpack_pack_str(&mp_pck, flb_sds_len(ctx->tag_key));
+            msgpack_pack_str_body(&mp_pck, ctx->tag_key, flb_sds_len(ctx->tag_key));
+            msgpack_pack_str(&mp_pck, tag_len);
+            msgpack_pack_str_body(&mp_pck, tag, tag_len);
+        }
+
+        msgpack_pack_str(&mp_pck, flb_sds_len(ctx->log_key));
+        msgpack_pack_str_body(&mp_pck, ctx->log_key, flb_sds_len(ctx->log_key));
+        msgpack_pack_object(&mp_pck, *obj);
+    }
+
+    /* Convert from msgpack to JSON */
+    out_buf = flb_msgpack_raw_to_json_sds(mp_sbuf.data, mp_sbuf.size);
+    msgpack_sbuffer_destroy(&mp_sbuf);
+
+    if (!out_buf) {
+        flb_plg_error(ctx->ins, "error formatting JSON payload");
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+
+    *out_data = out_buf;
+    *out_size = flb_sds_len(out_buf);
+
+    /* Cleanup */
+    msgpack_unpacked_destroy(&result);
+
+    return 0;
+}
+
+static void cb_azure_kusto_flush(struct flb_event_chunk *event_chunk,
+                                 struct flb_output_flush *out_flush,
+                                 struct flb_input_instance *i_ins, void *out_context,
+                                 struct flb_config *config)
+{
+    (void)i_ins;
+    (void)config;
+    int ret;
+    flb_sds_t json;
+    size_t json_size;
+    size_t tag_len;
+
+    struct flb_azure_kusto *ctx = out_context;
+
+    flb_plg_trace(ctx->ins, "flushing bytes %zu", event_chunk->size);
+
+    tag_len = flb_sds_len(event_chunk->tag);
+
+    /* Load or refresh ingestion resources */
+    ret = azure_kusto_load_ingestion_resources(ctx, config);
+    if (ret != 0) {
+        flb_plg_error(ctx->ins, "cannot load ingestion resources");
+        goto flush_error;
```

Comment:
Same as before, this jump could cause `json` to be accessed before it's set so setting it to `NULL` at the beginning of the function would be good. 

In this particular case this is a pretty simple function / flow so I'd really appreciate it if you could switch it not to use goto.

Reply:
Done, see my reply on the other comment